src/core/lib/security/credentials/s2a/grpc_s2a_credentials_options.h

```diff
@@ -31,42 +32,45 @@ struct grpc_s2a_credentials_options {
   ~grpc_s2a_credentials_options();
 
   /** Getters for member fields. **/
-  const char* handshaker_service_url() const { return handshaker_service_url_; }
+  const std::string handshaker_service_url() const {
+    return handshaker_service_url_;
+  }
   const std::vector<uint16_t>& supported_ciphersuites() const {
     return supported_ciphersuites_;
   }
-  const std::vector<char*>& target_service_account_list() const {
+  const std::vector<std::string>& target_service_account_list() const {
     return target_service_account_list_;
   }
 
   /** The setter method for |handshaker_service_url_|. This does not take
    *  ownership of the argument. **/
-  void set_handshaker_service_url(const char* handshaker_service_url);
+  void set_handshaker_service_url(std::string handshaker_service_url);
   /** This methods add |ciphersuite| to the vector |supported_ciphersuites_|; it
    *  does not remove duplicates from the vector, if they exist. See
    *  src/core/tsi/s2a/s2a_constants.h for the ciphersuite constants. **/
   void add_supported_ciphersuite(uint16_t ciphersuite);
   /** This API should only be called at the client-side, and any target service
    *  accounts that are added on the server-side will be ignored. This method
-   * adds a target service account to the vector |target_service_account_list_|;
-   * it does not remove duplicates from the vector, nor does it take ownership
-   * of the argument. **/
-  void add_target_service_account(const char* target_service_account);
+   *  adds a target service account to the vector
+   * |target_service_account_list_|; it does not remove duplicates from the
+   * vector, nor does it take ownership of the argument. **/
```

Comment:
Since the argument is not a pointer, we don't need to talk about ownership. Also, can we make the argument `const std::string&`?

Reply:
Yup, these changes have already been made! (This comment is on an intermediate commit for some reason and not on the most up-to-date commit.)